src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -47,21 +48,27 @@ def load_arguments(self, _):
     with self.argument_context('eventgrid topic') as c:
         c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
 
+    with self.argument_context('eventgrid topic key list') as c:
+        c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part=None, completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
```

Comment:
We set the id_part to None for this based on the discussion in https://github.com/Azure/azure-cli-extensions/pull/375#issuecomment-434072459. (In az eventgrid topic key list the presence of --ids to refer to the topic can be confusing because it is not the rightmost "resource" indicated by the command-group. We don't want --ids to specified for this case).

Reply:
@kalyanaj I meant that you don't need to specify arg_type, help, completer again.

Reply:
Got it, thanks.

Reply:
Fixed